gnocchi/rest/api.py

```diff
@@ -1892,6 +1900,67 @@ class BatchController(object):
     resources = ResourcesBatchController()
 
 
+class PrometheusWriteController(rest.RestController):
+
+    @staticmethod
+    @tenacity.retry(retry=tenacity.retry_if_exception_type(
+        (indexer.NoSuchResource, indexer.ResourceAlreadyExists,
+         indexer.NamedMetricAlreadyExists)))
+    def get_or_create_metrics(creator, rid, metric_names):
+        r = pecan.request.indexer.get_resource('generic', rid,
+                                               with_metrics=True)
+        if r:
+            exists_metric_names = [m.name for m in r.metrics]
+            metrics = MetricsSchema(dict(
+                (m, {}) for m in metric_names
+                if m not in exists_metric_names
+            ))
+            if metrics:
+                return pecan.request.indexer.update_resource(
+                    'generic', rid,
+                    metrics=metrics,
+                    append_metrics=True,
+                    create_revision=False
+                ).metrics
+            else:
+                return r.metrics
+        else:
+            metrics = MetricsSchema(dict((m, {}) for m in metric_names))
+            return pecan.request.indexer.create_resource(
+                'generic', rid, creator,
+                original_resource_id="prometheus",
+                metrics=metrics
+            ).metrics
+
+    @pecan.expose()
+    def post(self):
+        buf = snappy.uncompress(pecan.request.body)
+        f = remote_pb2.WriteRequest()
+        f.ParseFromString(buf)
+        measures = {}
+        for ts in f.timeseries:
+            attrs = dict((l.name, l.value) for l in ts.labels)
```

Comment:
I have reread the prometheus documentation and I have the feeling that Prometheus does uniquely identify metrics

I think putting all of them in "prometheus" resource will not help to query them.

Reply:
I have got how to identify a resource in prometheus. I will update the PR soon.